pkg/cmd/stack/deploy.go

```diff
@@ -698,19 +701,42 @@ func validateDefinedServices(s *model.Stack, servicesToDeploy []string) error {
 	return nil
 }
 
-func addDependentServicesIfNotPresent(ctx context.Context, s *model.Stack, options *StackDeployOptions, c kubernetes.Interface) {
-	added := make([]string, 0)
-	for _, svcToDeploy := range options.ServicesToDeploy {
+// AddDependentServicesIfNotPresent adds dependands services to deploy
+func AddDependentServicesIfNotPresent(ctx context.Context, s *model.Stack, svcsToDeploy []string, c kubernetes.Interface) []string {
+	initialSvcsToDeploy := svcsToDeploy
+	svcsToDeploy = addDependentServices(ctx, s, svcsToDeploy, c)
```

Comment:
Nit: maybe it is safer to create another variable instead of using the one passed as parameter `svcsToDeploy`